text_to_x/text_to_embeddings.py

```diff
@@ -0,0 +1,306 @@
+                
+    def find(self, query):
+        '''
+        input word, get vector representation
+        
+        Example:
+        In: self.find('hpv')
+        
+        Out: array([ 0.00435369, ...,  0.332069  ])
+        '''
+        if query in self.tok2ind:
+            query_index = self.tok2ind[query]
+            return self.model[query_index]
+        
+        raise ValueError("the word you're querying is not present in the model")
+        
+                
+                
+    def similar_to_query(self, query, n_similar = 5):
+        '''
+        print n most similar words to query (default 5)
+        cosine similarity is the metric used (min = -1, max = 1)
+        
+        - query: str, word to query
+        - n_similar: int, number of most simialr words to output
+        
+        Example:
+        In: self.similar_to_query('hpv')
+        
+        Out: [(0.3425330655678359, 'type'),
+             (0.29492709699932357, 'eksempel'),
+             (0.2840084462156332, 'larynxcance'),
+             (0.2736081170128576, 'sygdom'),
+             (0.24797964395624894, 'forårsag')]
+        '''
+        
+        if not isinstance(query, str):
+            raise ValueError(f"query expected a string")
+            
+        if not isinstance(n_similar, int):
+            raise ValueError(f"n_similar expected int")
+            
+        
+        vec = self.find(query)
+        
+        # cosine similarity
+        cos_sim_matrix = np.dot(self.model, vec)/(np.linalg.norm(self.model)*np.linalg.norm(vec))
+        
+        list_similar = []
+
+        for i in np.argpartition(-1 * cos_sim_matrix, n_similar + 1)[:n_similar + 1]:
+
+            if self.ind2tok[i] == query:
+                continue
+
+            list_similar.append((float(cos_sim_matrix[i]), self.ind2tok[i]))
+
+        return sorted(list_similar, reverse=True)
+    
+    
+    def similar_to_vec(self, vec, n_similar = 5):
+        '''
+        get most similar trained embeddings to an input vector
+        cosine similarity is the metric used (min = -1, max = 1)
+        
+        - vec: np.array, input vector (also a result of operation with vectors)
+        - n_similar: int, number of most simialr words to output
+        
+        Example:
+        In: self.similar_to_vec(vec = self.find('hpv') - self.find('pige'))
+        
+        Out: [(1.1030897070723877, 'hpv'),
+             (0.3426745031741564, 'type'),
+             (0.2964467662004768, 'eksempel'),
+             (0.2854145615758435, 'vort'),
+             (0.285052615742971, 'larynxcance'),
+             (0.27461291543657906, 'sygdom')]
+        '''
+        
+        # cosine similarity
+        cos_sim_matrix = np.dot(self.model, vec)/(np.linalg.norm(self.model)*np.linalg.norm(vec))
+        
+        list_similar = []
+
+        for i in np.argpartition(-1 * cos_sim_matrix, n_similar + 1)[:n_similar + 1]:
+
+            list_similar.append((float(cos_sim_matrix[i]), self.ind2tok[i]))
```

Comment:
Might be faster as list comprehension. E.g.:

```
list_similar = [(float(cos_sim_matrix[i]), self.ind2tok[i]) for i in \
                np.argpartition(-1 * cos_sim_matrix, n_similar + 1)[:n_similar + 1]]
```

Reply:
agree - again backslash is redundant